trio/_tests/test_threads.py

```diff
@@ -500,55 +540,57 @@ async def run_thread(event):
         c.total_tokens = orig_total_tokens
 
 
-async def test_run_in_worker_thread_custom_limiter():
+async def test_run_in_worker_thread_custom_limiter() -> None:
     # Basically just checking that we only call acquire_on_behalf_of and
     # release_on_behalf_of, since that's part of our documented API.
     record = []
 
     class CustomLimiter:
-        async def acquire_on_behalf_of(self, borrower):
+        async def acquire_on_behalf_of(self, borrower: Task) -> None:
             record.append("acquire")
             self._borrower = borrower
 
-        def release_on_behalf_of(self, borrower):
+        def release_on_behalf_of(self, borrower: Task) -> None:
             record.append("release")
             assert borrower == self._borrower
 
-    await to_thread_run_sync(lambda: None, limiter=CustomLimiter())
+    # TODO: should CapacityLimiter have an abc or protocol so users can modify it?
+    # because currently it's `final` so writing code like this is not allowed.
+    await to_thread_run_sync(lambda: None, limiter=CustomLimiter())  # type: ignore[arg-type]
```

Comment:
Obviously not something to do here, but I'd think it'd make most sense to do a protocol, with only the 6 `acquire_*`/`release_*` methods. That's all that a task needs to be limited, the others are only relevant for `CapacityLimiter`'s simple token strategy. The implementations of `acquire()`/`release()` could be included in the protocol since they defer to the `on_behalf_of` methods.

Reply:
Created #2824